### PR TITLE
input-field: various fixes

### PR DESCRIPTION
**Pull request by bvr-yr**
- correctly erase utf8 chars on backspace
- don't draw outer at all if `outline_thickness = 0`
- don't draw hidden input quadrants on fades. even if respecting `fade.a` and `data.opacity`, it's colored and still visible like quadrant (ugly)
- don't draw dots on unlock fadeout (they are flashing now)
- don't stuck on `check_color` on unlock fadeout
- now correctly checks `dots.currentAmount` against utf8 in draw return

I also made input-field fades to be a bit longer

### CODE REVIEW

**Comment by bvr-yr** [comment]
after c06afbe don't return on up events, instead just handle only down, but keep rendering

if keys pressed repeatedly, dots animation was kinda choppy
especially when pressing same key twice fast enough
I have two same characters at the end and very often last dot didn't render, but only after few seconds it did

haven't noticed any performance difference even on my shitty gpu

File: src/renderer/widgets/PasswordInputField.cpp [hunk 1]
```diff
@@ -124,9 +122,9 @@ void CPasswordInputField::updateFade() {
 
     if (fade.animated) {
         if (fade.appearing)
-            fade.a = std::clamp(std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - fade.start).count() / 100000.0, 0.0, 1.0);
+            fade.a = std::clamp(std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - fade.start).count() / 150000.0, 0.0, 1.0);
         else
-            fade.a = std::clamp(1.0 - std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - fade.start).count() / 100000.0, 0.0, 1.0);
+            fade.a = std::clamp(1.0 - std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - fade.start).count() / 150000.0, 0.0, 1.0);
```
Comment (vaxerski) [inline src/renderer/widgets/PasswordInputField.cpp:127]: why this change?

File: src/renderer/widgets/PasswordInputField.cpp [hunk 2]
```diff
@@ -175,6 +171,9 @@ bool CPasswordInputField::draw(const SRenderData& data) {
 
     bool forceReload = false;
 
+    passwordLength = g_pHyprlock->getPasswordBufferDisplayLen();
```
[the remark this answers is not in the document]
Comment (vaxerski) [reply src/renderer/widgets/PasswordInputField.cpp:174]: its fine

File: src/renderer/widgets/PasswordInputField.cpp (hunk 1 above)
Comment (bvr-yr) [reply src/renderer/widgets/PasswordInputField.cpp:127]: seemed to quick, just to line-up with other animations. change it back?

Comment (vaxerski) [reply src/renderer/widgets/PasswordInputField.cpp:127]: yea

**Review by vaxerski — APPROVED** [review]
